packages/typo3-docs-theme/resources/js/components/SearchModal.jsx

```diff
@@ -0,0 +1,276 @@
+import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import { useSearchScopes } from '../hooks/useSearchScopes';
+import { useSearchSuggestions } from '../hooks/useSearchSuggestions';
+import SuggestRow from './SuggestRow';
+import { PROXY_URL } from '../search';
+
+const SearchModal = ({ isOpen, onClose }) => {
+    const [searchQuery, setSearchQuery] = useState('');
+    const [scopes, setScopes] = useSearchScopes();
+    const [activeIndex, setActiveIndex] = useState(-1);
+    const suggestionsRef = useRef([]);
+    const inputRef = useRef();
+
+    const {
+        fileSuggestions,
+        scopeSuggestions,
+        isLoading,
+        fetchSuggestions
+    } = useSearchSuggestions();
+
+    const buildHref = useCallback((scopes, query) => {
+        const url = new URL('/search/search', PROXY_URL);
+        scopes.forEach(scope => {
+            if (scope.type === 'manual' || scope.type === 'vendor') {
+                url.searchParams.append('scope', encodeURIComponent(`/${scope.slug}/`));
+            } else if (scope.type === 'option') {
+                url.searchParams.append(`filters[optionaggs][${scope.title}]`, true);
+            } else {
+                url.searchParams.append(`filters[${scope.type}][${scope.title}]`, true);
+            }
+        });
+        url.searchParams.append('q', query);
+        return url.href;
+    }, []);
```

Comment:
We miss something here.

In suggestions, we need the manual_slug, to build proper contextualized urls.

Reply:
Done :white_check_mark: